mono/utils/lock-free-alloc.c

```diff
@@ -87,7 +87,16 @@ enum {
 	STATE_EMPTY
 };
 
-typedef union {
+typedef union Anchor {
+
+// C++ does not by default allow assigning volatile aggregates (struct, union, class).
+#ifdef __cplusplus
+	Anchor& operator=(const volatile Anchor& other)
```

Comment:
No C++ allowed.

Reply:
Better now with https://github.com/mono/mono/pull/9819/commits/05d19341fd2d8f30c3a14d06b17f395a722277ba ?
Feels a bit abstraction boundary violating -- better to assign aggregate than members.
Maybe write a function/macro, `mono_anchor_assign_volatile`?